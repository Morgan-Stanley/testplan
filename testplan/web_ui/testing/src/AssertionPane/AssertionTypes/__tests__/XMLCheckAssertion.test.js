import React from "react";
import { shallow } from "enzyme";
import { StyleSheetTestUtils } from "aphrodite";
import XMLCheckAssertion from "../XMLCheckAssertion";

describe("XMLChekAssertion", () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  it("shallow renders", () => {
    const props = {
      assertion: {
        category: "DEFAULT",
        machine_time: "2019-02-12T17:41:43.321541+00:00",
        description: "Simple XML check for existence of xpath.",
        xpath: "/Root/Test",
        xml: "<Root>\n                <Test>Foo</Test>\n            </Root>\n",
        data: [],
        line_no: 710,
        tags: null,
        meta_type: "assertion",
        passed: true,
        message: "xpath: `/Root/Test` exists in the XML.",
        type: "XMLCheck",
        namespaces: null,
        utc_time: "2019-02-12T17:41:43.321534+00:00",
      },
    };
    const shallowComponent = shallow(<XMLCheckAssertion {...props} />);
    expect(shallowComponent).toMatchSnapshot();
  });
});
