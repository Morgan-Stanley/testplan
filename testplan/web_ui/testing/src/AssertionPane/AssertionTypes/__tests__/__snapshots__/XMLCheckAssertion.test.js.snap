// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XMLChekAssertion shallow renders 1`] = `
<Fragment>
  <div>
    <strong
      className=""
    >
      Expected XPath:
    </strong>
    <span>
      /Root/Test
    </span>
  </div>
  <pre
    className="inputText_197f7f1"
  >
    &lt;Root&gt;
    &lt;Test&gt;Foo&lt;/Test&gt;
&lt;/Root&gt;

  </pre>
</Fragment>
`;
